x/bsonx/bsoncore/array.go

```diff
@@ -81,18 +94,16 @@ func (a Array) String() string {
 
 	var elem Element
 	var ok bool
-	first := true
 	for length > 1 {
-		if !first {
-			buf.WriteByte(',')
-		}
 		elem, rem, ok = ReadElement(rem)
 		length -= int32(len(elem))
 		if !ok {
 			return ""
 		}
-		fmt.Fprintf(&buf, "%s", elem.String())
-		first = false
+		fmt.Fprintf(&buf, "%s", elem.Value().String())
+		if length != 1 {
+			buf.WriteByte(',')
+		}
```

Comment:
Similarly, it might be worth returning "" if length != 1 after the loop, since that would be an invalid array.

Reply:
Added that in, too! Note that I won't add the same checks for `DebugString()` since it makes no promise of validating the array, and it might be helpful for users to see some piece of their malformed array printed out.